### PR TITLE
prevent toast message appearing when creating new module instance

### LINKED ISSUE #4506: create new module instance - "raster mask:no mask used" toast appears
On current master, create a new or duplicate instance of a module.

Toast message appears: "raster mask: no mask used"

This is caused by https://github.com/darktable-org/darktable/commit/12041936d0d85a0b7b077c08d242d7fdee240d28

### PR DESCRIPTION
**Pull request by elstoc**
Resolves #4506

### CODE REVIEW

File: src/develop/imageop.c [hunk 1]
```diff
@@ -813,6 +813,7 @@ dt_iop_module_t *dt_iop_gui_duplicate(dt_iop_module_t *base, gboolean copy_param
   /* initialize gui if iop have one defined */
   if(!dt_iop_is_hidden(module))
   {
+    darktable.gui->reset = 1;
```
Comment (TurboGit) [inline src/develop/imageop.c]: before you need to had:
`const int reset = darktable.gui->reset;`

and...

File: src/develop/imageop.c [hunk 2]
```diff
@@ -844,6 +845,7 @@ dt_iop_module_t *dt_iop_gui_duplicate(dt_iop_module_t *base, gboolean copy_param
                           expander, g_value_get_int(&gv) + pos_base - pos_module + 1);
     dt_iop_gui_set_expanded(module, TRUE, FALSE);
     dt_iop_gui_update_blending(module);
+    darktable.gui->reset = 0;
```
Comment (TurboGit) [inline src/develop/imageop.c]: darktable.gui->reset = reset;

To properly reset the state as it was entering this routine.

**Comment by elstoc** [comment]
Thanks @TurboGit. In my defence it was late!

**Comment by TurboGit** [comment]
Thanks to you @elstoc , and no need for defense we all do mistakes :) The important point is to fix them at the right time. Thanks for you efforts.

**Comment by TurboGit** [comment]
We have another similar issue when changing items in the tools at the bottom-right of the lighttable and darkroom (screen profile, intent...). Each time you switch settings a toast is displayed. Can you have a look? Thanks.

**Comment by elstoc** [comment]
will do

**Comment by elstoc** [comment]
I'm sorry after a few hours fruitless investigation I can't work out where this is coming from. I've tried putting gui resets before all of the calls to `dt_bauhaus_combobox_set` in the lighttable and darkroom views and the toasts are still appearing.

**Pull request merged**